lib/sender_link.js

```diff
@@ -75,10 +73,15 @@ SenderLink.prototype.send = function(msg, options) {
   return new Promise(function(resolve, reject) {
     var message;
     if (u.isObject(msg) && msg.hasOwnProperty('body')) {
-      message = new M.Message(msg);
+      message = msg;
       message.body = self.policy.encoder(message.body);
     } else {
-      message = new M.Message(options, self.policy.encoder(msg));
+      if (u.isObject(options)) {
+        options.body = self.policy.encoder(msg);
```

Comment:
I realize this is a pre-existing bug, but should we protect against missing self.policy.encoder? We do against this.policy.decoder in receiver.